### ISSUE TITLE
Doesn't load project plugins

### ISSUE THREAD

**Issue by tmoitie** [codeclimate/codeclimate-eslint#10]
My eslintrc file is trying to use the react plugin which is not loaded by the analyzer engine

`error: (CC::Analyzer::Engine::EngineFailure) engine eslint failed with status 1 and stderr "module.js:338\n    throw err;\n          ^\nError: Cannot find module 'eslint-plugin-react'\n    at Function.Module._resolveFilename (module.js:336:15)\n    at Function.Module._load (module.js:278:25)\n    at Module.require (module.js:365:17)\n    at require (module.js:384:17)\n    at /usr/src/app/node_modules/eslint/lib/cli-engine.js:107:26\n    at Array.forEach (native)\n    at loadPlugins (/usr/src/app/node_modules/eslint/lib/cli-engine.js:98:21)\n    at processFile (/usr/src/app/node_modules/eslint/lib/cli-engine.js:179:5)\n    at /usr/src/app/node_modules/eslint/lib/cli-engine.js:332:26\n    at /usr/src/app/node_modules/eslint/lib/util/traverse.js:61:17\n"
`

**Comment by GordonDiggs** [comment]
Thanks! I'm putting this in our roadmap and will fix soon

**Comment by tmoitie** [comment]
Hullo. Has this PR made it into release? If so how do I update?

Thanks!

**Comment by GordonDiggs** [comment]
@tmoitie We had to revert it because we actually can't support plugins just yet. I'm going to leave this issue open for now and will update it when we push support for them!

**Comment by jdelStrother** [comment]
In the meantime, any suggestions on continuing to use `eslint-plugin-react` locally, but disabling it when running under codeclimate?   Can I point codeclimate's engine to a different .eslintrc or something?

**Comment by jakemhiller** [comment]
Yeah, Having the same issue. Along with the react plugin we also use the [babel-eslint](https://github.com/babel/babel-eslint) parser, which was mentioned in another issue. It would be nice to at least have a work-around like what @jdelStrother mentioned.

**Comment by mattdaw** [comment]
:+1: My project is blocked by this as well.

**Comment by abritinthebay** [comment]
Ditto. At the very least supporting the react plugin will allow support of all the react code coming down the pipe.

While ESlint killed support of React semantics in JSX (which makes sense, tho I wish they'd just have a JSX plugin at that point) this is basically required for any JSX work that uses components (ie - most of it)

**Comment by GordonDiggs** [comment]
Hey all! We are now bundling the `eslint-plugin-react`, `eslint-plugin-babel` and `babel-eslint` packages with this engine, so you should be able to use them moving forward!

**Comment by GordonDiggs** [comment]
We hope to bring more general support for plugins in the future

**Comment by kachkaev** [comment]
Hey guys,

Any updates in your plans on eslint plugin support?

No worries if the feature can't be done in the nearest future. In this case, it would be nice if the issue was reflected in the docs somehow. It took me a pretty long while to discover the basis of an error I saw in the codeclimate report. To be more precise, I was googling for ```codeclimate "Definition for rule" "was not found"``` but that did not return anything informative.

**Comment by averyvery** [comment]
Tagging onto this; we have a project eslinting with the `jsx-control-statements` plugin (and extended ruleset) and it would be great if Code Climate could pick this up when linting our JS.

**Comment by pgoldrbx** [comment]
Adding a bump here — specifically, `eslint-plugin-flowtype` (though, tbh, I may move away from flow in favor of more robust tests)

**Comment by athyuttamre** [comment]
+1 I'm trying to use `eslint-plugin-meteor` and can't get CodeClimate working.

**Comment by maxjacobson** [comment]
@athyuttamre Hi there. You can always check [our eslint docs](https://docs.codeclimate.com/docs/eslint#section-supported-plugins) to see the currently-supported plugins. If you'd like us to add support for another plugin, please feel free to open an issue telling us about it.

**Comment by athyuttamre** [comment]
Done, thank you! https://github.com/codeclimate/codeclimate-eslint/issues/155
